modules/thunder_updater/thunder_updater.install

```diff
@@ -5,23 +5,105 @@
+  /** @var \Drupal\Core\Extension\ModuleHandlerInterface $module_handler */
+  $module_handler = \Drupal::moduleHandler();
 
-  // Install Checklist API as soft dependency.
-  try {
-    \Drupal::service('module_installer')->install(['checklistapi']);
-    \Drupal::service('thunder_updater.update_checklist')->markAllUpdates();
-  }
-  catch (MissingDependencyException $e) {
-    $logger->warning(
-      'Could not install Checklist API module. You will not see information 
-      about updates in the drupal admin pages.'
-    );
+  /** @var \Drupal\Core\Extension\ModuleInstallerInterface $module_installer */
+  $module_installer = \Drupal::service('module_installer');
+
+  /** @var \Drupal\Core\Config\ConfigFactoryInterface $config_factory */
+  $config_factory = \Drupal::configFactory();
+
+  // We let exception bubble in this case, since it's not possible to continue
+  // with execution of update hooks without "update_helper.updater" service.
+  $module_installer->install(['update_helper']);
+
+  /** @var \Drupal\update_helper\Updater $updater */
+  $updater = \Drupal::service('update_helper.updater');
+  $update_logger = $updater->logger();
+
+  // We need database connection in order to migrate Thunder updates correctly.
+  $database = \Drupal::database();
+
+  // Existing module states should be preserved!
+  $has_thunder_updater_checklist = $module_handler->moduleExists('checklistapi');
+
+  // Migrate existing Thunder update entities and checklist config.
+  if ($has_thunder_updater_checklist) {
+    try {
+      $module_installer->install(['update_helper_checklist']);
+
+      /** @var \Drupal\update_helper_checklist\UpdateChecklist $update_checklist */
+      $update_checklist = \Drupal::service('update_helper_checklist.update_checklist');
+      $update_checklist->markAllUpdates(FALSE);
+
+      $update_checklist->markUpdatesSuccessful(['thunder' => ['thunder_updater_update_8001']]);
+    }
+    catch (Exception $e) {
+      $update_logger->warning('Unable to install "update_helper_checklist".');
+    }
+
+    // We do not need entities created on "update_helper_checklist" install.
+    /** @var \Drupal\update_helper_checklist\Entity\Update[] $all_updates */
+    $all_updates = Update::loadMultiple();
+    foreach ($all_updates as $update) {
+      try {
+        // We want to keep status for this update hook.
+        if ($update->id() === 'thunder:thunder_updater_update_8001') {
+          continue;
+        }
+
+        $update->delete();
+      }
+      catch (EntityStorageException $e) {
+        $update_logger->warning(sprintf('Unable to delete Update entity (%s) for Update Helper Checklist.', $update->id()));
+
+        $update_checklist->markUpdatesFailed(['thunder' => ['thunder_updater_update_8001']]);
+      }
+    }
+
+    $table_name = $database->getFullQualifiedTableName('thunder_updater_update');
+    $thunder_updater_updates = $database->query("SELECT * FROM {$table_name};")->fetchAll(PDO::FETCH_ASSOC);
```

Comment:
why all the hassle? The {thunder_updater_update} in the query takes care of the correct table name

Reply:
👍